include/unifex/indexed_for.hpp

```diff
@@ -74,39 +74,183 @@ struct indexed_for_sender {
     return blocking(sender.pred_);
   }
 
+  // sequenced_policy version supports forward range
+  template<typename... Values>
+  static void apply_func_with_policy(const execution::sequenced_policy& policy, Range&& range, Func&& func, Values&... values)
+      noexcept(std::is_nothrow_invocable_v<Func, typename Range::iterator::value_type, Values...>) {
+    for(auto idx : range) {
+      std::invoke(func, idx, values...);
+    }
+  }
+
+  // parallel_policy version requires random access range
+  template<typename... Values>
+  static void apply_func_with_policy(const execution::parallel_policy& policy, Range&& range, Func&& func, Values&... values)
+      noexcept(std::is_nothrow_invocable_v<Func, typename Range::iterator::value_type, Values...>) {
+    auto start = range.begin();
+    for(auto idx = 0; idx < range.size(); ++idx) {
+      std::invoke(func, start[idx], values...);
+    }
+  }
+
   template <typename Receiver>
   struct indexed_for_receiver {
     UNIFEX_NO_UNIQUE_ADDRESS Func func_;
     UNIFEX_NO_UNIQUE_ADDRESS Policy policy_;
     UNIFEX_NO_UNIQUE_ADDRESS Range range_;
     UNIFEX_NO_UNIQUE_ADDRESS Receiver receiver_;
 
-    // sequenced_policy version supports forward range
-    template<typename... Values>
-    static void apply_func_with_policy(const execution::sequenced_policy& policy, Range&& range, Func&& func, Values&... values)
-        noexcept(std::is_nothrow_invocable_v<Func, typename std::iterator_traits<typename Range::iterator>::reference, Values...>) {
-      for(auto idx : range) {
-        std::invoke(func, idx, values...);
+    template <typename... Values>
```

Comment:
No wonder the rebase messed up. Clearly this diff algorithm got very confused with this similar-but-different code.